src/test/java/com/gluonhq/substrate/model/InternalProjectConfigurationTest.java

```diff
@@ -71,4 +71,67 @@ static Stream<Arguments> versioningSchemeParameters() {
                 Arguments.of("22.0.2", false)
         );
     }
+
+    @ParameterizedTest
+    @MethodSource("graalVersionOutputs")
+    void testParseGraalVMVersion(Version graalVersion, Version javaVersion, String output) {
+        assertEquals(graalVersion.toString(), InternalProjectConfiguration.parseGraalVersion(output).toString());
+    }
+
+    @ParameterizedTest
+    @MethodSource("graalVersionOutputs")
+    void testParseGraalVMJavaVersion(Version graalVersion, Version javaVersion, String output) {
+        assertEquals(javaVersion.toString(), InternalProjectConfiguration.parseGraalVMJavaVersion(output).toString());
+    }
+
+    static Stream<Arguments> graalVersionOutputs() {
+        // outputs of 'java -version'
+        return Stream.of(
+                // ======== Gluon Builds ========
+
+                // graalvm-svm-java17-linux-gluon-22.1.0.1-Final
+                Arguments.of(new Version("22.1.0"), new Version("17.0.3"), "openjdk version \"17.0.3\" 2022-04-19\n" +
+                        "OpenJDK Runtime Environment GraalVM 22.1.0.1 (build 17.0.3+7-jvmci-22.1-b06)\n" +
+                        "OpenJDK 64-Bit Server VM GraalVM 22.1.0.1 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing)"),
+
+                // graalvm-svm-java17-windows-gluon-22.0.0.3-Final
+                Arguments.of(new Version("22.0.0"), new Version("17.0.2"), "openjdk version \"17.0.2\" 2022-01-18\n" +
+                        "OpenJDK Runtime Environment GraalVM 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05)\n" +
+                        "OpenJDK 64-Bit Server VM GraalVM 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05, mixed mode, sharing)"),
+
+                // graalvm-svm-java17-windows-gluon-22.1.0.1-Final
+                Arguments.of(new Version("22.1.0"), new Version("17.0.3"), "openjdk version \"17.0.3\" 2022-04-19\n" +
+                        "OpenJDK Runtime Environment GraalVM 22.1.0.1 (build 17.0.3+7-jvmci-22.1-b06)\n" +
+                        "OpenJDK 64-Bit Server VM GraalVM 22.1.0.1 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing)"),
+
+                // ======== Oracle Builds ========
```

Comment:
This needs to include tests for community edition as well
```
openjdk 17.0.7 2023-04-18
OpenJDK Runtime Environment GraalVM CE 17.0.7+7.1 (build 17.0.7+7-jvmci-23.0-b12)
OpenJDK 64-Bit Server VM GraalVM CE 17.0.7+7.1 (build 17.0.7+7-jvmci-23.0-b12, mixed mode, sharing)
```

```
openjdk 21 2023-09-19
OpenJDK Runtime Environment GraalVM CE 21+35.1 (build 21+35-jvmci-23.1-b15)
OpenJDK 64-Bit Server VM GraalVM CE 21+35.1 (build 21+35-jvmci-23.1-b15, mixed mode, sharing)
```

Reply:
Good catch. The `.*?` got removed by accident. Fixed and added unit tests.